config/core/default_env_patch.yaml

```diff
@@ -0,0 +1,18 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: controller-manager
+  namespace: system
+spec:
+  template:
+    spec:
+      containers:
+      - name: manager
+        env:
+        - name: DISABLE_DEFAULT_ARGOCD_INSTANCE
+          value: "true"
+        - name: DISABLE_DEFAULT_ARGOCD_CONSOLELINK
+          value: "true"
+        - name: DISABLE_DEX
```

Comment:
I think we need to disable KAM deployments too.

Reply:
Hi @ishitasequeira! According to the [documentation](https://github.com/redhat-developer/gitops-operator/blob/master/docs/OpenShift%20GitOps%20Usage%20Guide.md#setting-environment-variables), there is no way to disable KAM by using the container Environment.
If the Core operator cannot provide this mechanism, I'm afraid that it would need to be either amended or shipped along with KAM components.

Reply:
As discussed over chat, let's create a separate story for `DISABLE_KAM` flag and get this PR merged.